instrumentation/dubbo/dubbo-apache-2.7/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/dubbo/apache/v2_7/DubboFilterInstrumentation.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.dubbo.apache.v2_7;
+
+import static io.opentelemetry.javaagent.tooling.bytebuddy.matcher.ClassLoaderMatcher.hasClassesNamed;
+import static java.util.Collections.singletonMap;
+import static net.bytebuddy.matcher.ElementMatchers.isPublic;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+
+import io.opentelemetry.instrumentation.dubbo.apache.v2_7.client.TracingClientFilter;
+import io.opentelemetry.instrumentation.dubbo.apache.v2_7.server.TracingServerFilter;
+import io.opentelemetry.javaagent.tooling.TypeInstrumentation;
+import java.util.List;
+import java.util.Map;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.apache.dubbo.rpc.Filter;
+
+public class DubboFilterInstrumentation implements TypeInstrumentation {
+
+  @Override
+  public ElementMatcher<ClassLoader> classLoaderOptimization() {
+    return hasClassesNamed("org.apache.dubbo.common.extension.ExtensionLoader");
+  }
+
+  @Override
+  public ElementMatcher<? super TypeDescription> typeMatcher() {
+    return named("org.apache.dubbo.common.extension.ExtensionLoader");
+  }
+
+  @Override
+  public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
+    return singletonMap(
+        named("getActivateExtension")
+            .and(takesArgument(0, named("org.apache.dubbo.common.URL")))
+            .and(takesArgument(1, String.class))
+            .and(takesArgument(2, String.class))
+            .and(isPublic()),
+        DubboFilterInstrumentation.class.getName() + "$AddFilterAdvice");
```

Comment:
Does it work to use resource injection instead?

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/instrumentation/aws-sdk/aws-sdk-2.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/awssdk/v2_2/AwsSdkInstrumentationModule.java#L33

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/instrumentation/aws-sdk/aws-sdk-2.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/awssdk/v2_2/AwsSdkInitializationInstrumentation.java#L23

Reply:
In dubbo, I found that importing resources directly does not work.It should be declared or configured in advance.Maybe code here it's easier to understand now。

Reply:
Can you describe more what the issue is? This injection happens on classload, so it's as early as it needs to be I think. For the type matcher in `DubboInitializationInstrumentation`, I think you can just use `ExtensionLoader`.

Looking at this version of the instrumentation, I see we instrument one of these similarly named methods

https://github.com/apache/dubbo/blob/master/dubbo-common/src/main/java/org/apache/dubbo/common/extension/ExtensionLoader.java#L246

It's unclear whether this is complete, or if it will need maintenance in the future. We can be more confident if we use the official mechanism though, which I guess is the resource file?